File: exercism/javascript/phone-number/phone-number.js
class PhoneNumber {
  constructor(digits) {
    this.digits = digits;
  }

  number() {
    let numbers = this.digits.match(/[0-9]/g);
    if (this.digits.match(/[a-z]/i)) {
      return null;
    } else if (numbers.length === 10) {
      return numbers.join('');
    } else if (numbers[0] == 1 && numbers.length === 11) {
      return numbers.slice(1).join('');
    } else {
      return null;
    }
  }
}

export default PhoneNumber;

File: exercism/ruby/complete/phone-number/phone_number.rb
class PhoneNumber
  def self.clean(string)
    numbers = string.scan(/[0-9]/)
    n = numbers.length
    if n == 10 || n == 11
      return nil if n == 11 && numbers[0].to_i != 1
      numbers = numbers.last(10)
      numbers.join if numbers[0].to_i > 1 && numbers[3].to_i > 1
    end
  end
end
